monkey/common/network/network_range.py

```diff
@@ -77,7 +80,21 @@ def validate_range(address_str: str):
             raise InvalidNetworkRangeError(e)
 
     @staticmethod
-    def check_if_range(address_str):
+    def check_if_hostname(hostname: str):
+        if len(hostname) > 253 or hostname[-1] == ".":
+            return False
+
+        labels = hostname.split(".")
+
+        # the TLD must be not all-numeric
+        if re.match(r"[0-9]+$", labels[-1]):
+            return False
+
+        allowed = re.compile(r"(?!-)[a-z0-9-]{1,63}(?<!-)$", re.IGNORECASE)
```

Comment:
This regex can be compiled once and stored in this module or as a class variable, rather than every time the function is called.